### DISCUSSION TITLE
socksify ip address

### DISCUSSION THREAD

**Discussion by steom** [wiresock/proxifyre#5] [Ideas]
please consider to support also destinations ip address directly
so we can socksify the entire system and not just "appNames"

AllowIP: 0.0.0.0/0
DisallowIP: 192.168.10.0/20, 10.100.50.64/32

**Comment by wiresock** [comment]
I've been contemplating your idea. Regarding the configuration file, how do you envision its structure in this context? Should the 'allowedips' be specified in each proxy block? How would this integrate with application names? While I could adopt the strategy used in the WireSock VPN Client, the complexity may increase with the introduction of multiple proxies. I would appreciate your insights on this matter.

**Comment by steom** [comment]
in additions to filter for "appNames", for every proxy blocks, implements also:

"Routes": packects send to specified DNS names and ip addresses should connect to the ProxyEndpoint

"Ports": packects send to matching destinations ports should connect to the ProxyEndpoint

"localsubnets": packects send to specified DNS names and ip addresses should **never** connect to the ProxyEndpoint

these 'filters' can be used individually or together
